### ISSUE TITLE
Invasion: some planets cannot be invaded

### ISSUE THREAD

**Issue by Grummel7** [freeorion/freeorion#4083]
Found with 555e940e0a56003c91d11279515fb977ff939070. I think I've seen this before, but I do not play may games myself, I mostly let the AIs battle each other...
Has nobody else observed it?

![invasion_issue](https://user-images.githubusercontent.com/20499830/184818562-ac74db95-9df8-4a3b-9cbb-9d24e3984443.png)
I just attacked a system successfully, all planets were incapacitated, but two have a partially recovered shield and cannot be invaded.

I advanced a few turns and the result was always the same: It was not always planet II that was incapacitated last, but it always only planet II was could be invaded. 

I loaded other games and did a few tries, I found cases where 1 of 5 or even no planet was invadable at all.

Save file attached.
[cannot_invade.sav.gz](https://github.com/freeorion/freeorion/files/9348651/cannot_invade.sav.gz)

**Comment by agrrr3** [comment]
in theory, the focs effect for regrowing shields could be confused

**Comment by agrrr3** [comment]
one thing which did change not so long ago is the move to python FOCS for the defense techs. Maybe LocalCandidate.LastTurnAttackedByShip works differently there?
but @o01eg made sure the resulting objects have the same checksum as the ones created by the text FOCS files, so this is unlikely

**Comment by o01eg** [comment]
You could check if I mentioned checksum changed when converting related techs.

**Comment by o01eg** [comment]
Yes, it had the same checksum https://github.com/freeorion/freeorion/pull/3845

**Comment by Grummel7** [comment]
It's not a python problem.

I've debugged a bit and suddenly one of the planets got shields 16. Then, by modifying the scripted increases, I found that both the effects from DEF_ROOT_DEFENSE (Defense.focs.py) and STANDARD_SHIELD_GROWTH (planet_shields.macros) are triggered, although both require (LocalCandidate.LastTurnConquered < CurrentTurn). 

Note that I also found that the Barrier Shields actually check (LocalCandidate.LastTurnConquered < CurrentTurn - 1). Which explains why the bigger increase only becomes effective 2 turns after an attack. Is this intended?
